pkg/reconciler/route/route.go

```diff
@@ -302,7 +302,7 @@ func (c *Reconciler) reconcileIngressResources(ctx context.Context, r *v1alpha1.
 
 func (c *Reconciler) tls(ctx context.Context, host string, r *v1alpha1.Route, traffic *traffic.Config, clusterLocalServiceNames sets.String) ([]netv1alpha1.IngressTLS, error) {
 	tls := []netv1alpha1.IngressTLS{}
-	if !config.FromContext(ctx).Network.AutoTLS {
+	if !config.FromContext(ctx).Network.AutoTLS || r.Labels[config.VisibilityLabelKey] == config.VisibilityClusterLocal {
```

Comment:
I think the better change is to do something similar to line 314 by checking if the Route Domain is cluster local.  That would cover the case where the main Route's placeholder Service is labeled with cluster-local value.

Perhaps we can change tagToDomainMap to allow `""` tag to mean the main Route domain and remove code where where we treat the Route domain as a special case.

Reply:
Thank you. Yes, you are correct. I confirmed that individual svc could be a cluster local, and it also created the certs.
Then, after some researching the code, the Line 314 (especially line 316) looks wrong as `tagToDomainMap`'s key is domain not tag. So, I fixed it and seems to cover all cases.

Reply:
Thanks a lot!  Two birds with one stone.